tests/sentry/incidents/test_charts.py

```diff
@@ -128,30 +128,44 @@ def test_eap_alert(self, mock_client_get: MagicMock, mock_generate_chart: MagicM
 
 class FetchOpenPeriodsTest(TestCase):
     @freeze_time(frozen_time)
-    @patch("sentry.incidents.charts.client.get")
+    @with_feature("organizations:incidents")
     @with_feature("organizations:workflow-engine-single-process-metric-issues")
-    def test_get_incidents_from_detector(self, mock_client_get: MagicMock) -> None:
+    def test_get_incidents_from_detector(self) -> None:
         self.create_detector()  # dummy so detector ID != alert rule ID
         detector = self.create_detector(project=self.project)
         alert_rule = self.create_alert_rule(organization=self.organization, projects=[self.project])
         self.create_alert_rule_detector(detector=detector, alert_rule_id=alert_rule.id)
-        incident = Incident(
+        incident = self.create_incident(
             date_started=must_parse_datetime("2022-05-16T18:55:00Z"),
-            date_closed=None,
+            status=IncidentStatus.CRITICAL.value,
             alert_rule=alert_rule,
         )
+        # create incident activity the same way we do in logic.py create_incident
+        detected_activity = self.create_incident_activity(
+            incident,
+            IncidentActivityType.DETECTED.value,
+            date_added=incident.date_started,
+        )
+        created_activity = self.create_incident_activity(
+            incident,
+            IncidentActivityType.CREATED.value,
+        )
+
         time_period = incident_date_range(60, incident.date_started, incident.date_closed)
 
-        fetch_metric_issue_open_periods(self.organization, detector.id, time_period)
-        mock_client_get.assert_called_with(
```

Comment:
I recognize that I'm changing what's being tested here, but I'm not sure the previous test was adding much value. I can keep it around and add this as an additional test if necessary.